pkg/status/status.go

```diff
@@ -81,17 +90,17 @@ func New(cfg *rest.Config, mgr manager.Manager, namespace string, version string
 }
 
 // SetFailing reports that operator has failed along with the error message
-func (s *status) SetFailing(message string) {
+func (s *status) setFailing(message string) {
 	s.setStatus(configv1.OperatorFailing, message)
 }
 
 // SetAvailable reports that the operator is available to process events
-func (s *status) SetAvailable(message string) {
+func (s *status) setAvailable(message string) {
 	s.setStatus(configv1.OperatorAvailable, message)
 }
 
 // SetProgressing reports that the operator is being deployed
-func (s *status) SetProgressing(message string) {
+func (s *status) setProgressing(message string) {
 	s.setStatus(configv1.OperatorProgressing, message)
 }
```

Comment:
At this point we do not allow setting multiple conditions. This becomes an issue when v1 of the operator is installed and available but v2 is being rolled out. Technically the CO should report that the operator is available for version v1 and is progressing to v2. Let me know your thoughts.